src/org/sosy_lab/java_smt/solvers/yices2/Yices2NativeApiTest.java

```diff
@@ -580,4 +593,27 @@ public void bvMul() {
     System.out.println(component[1]);
     System.out.println(yices_term_constructor(yices_bvpower(component[0], component[1])));
   }
+
+  /**
+   * Only to be used for tests in this class. Old implementation used for creating/retrieving named
+   * variables. Superseded by Yices2FormulaCreator.createNamedVariable() for reasons outlined there.
+   */
+  private static int yices_named_variable(int type, String name) {
+    int termFromName = yices_get_term_by_name(name);
+    if (termFromName != -1) {
+      int termFromNameType = yices_type_of_term(termFromName);
+      if (type == termFromNameType) {
+        return termFromName;
+      } else {
+        throw new IllegalArgumentException(
+            String.format(
+                "Can't create variable with name '%s' and type '%s' "
+                    + "as it would omit a variable with type '%s'",
+                name, yices_type_to_string(type), yices_type_to_string(termFromNameType)));
+      }
+    }
+    int var = yices_new_uninterpreted_term(type);
+    yices_set_term_name(var, name);
+    return var;
+  }
```

Comment:
This IF-THEN-ELSE logic seems to be used twice. Can we refactor it into a proper utility method?

Reply:
Only parts of the structure is reused. This version is for tests and for free variables only. The other one is for bound variables and has additional logic. (I reworked the creation of free vars and its not similar to this one anymore)